tt.sh

```diff
@@ -35,13 +35,24 @@ function time_in {
 		time="$(date '+%Y-%m-%d') ${1}:00"
 		activity="${@:2}"
 	fi
-	echo "Begin ${activity} at ${time}"
-	echo "i ${time} ${activity}" >> $TIME_FILE
+	#
+	# Split the activity into:
+	#
+	#   - the project: anything before the first whitespace(s)
+	#     whitespace-containing project name aren't supported.
+	#     (maps to the hledger transaction virtual account)
+	#
+	#   - the project task: anything following the first whitespace(s)
+	#     (maps to the hledger transaction description)
+	#
+	local project_task=$(echo "${activity}" | sed -E "s/[[:blank:]]+/  /")
+	echo "Begin ${project_task} at ${time}"
+	echo "i ${time} ${project_task}" >> $TIME_FILE
 }
 
 function time_out {
 	local time=$(date '+%Y-%m-%d %H:%M:%S')
-	if [[ "${1}" =~ ^[0-9]{2}:[0-9]{2}$ ]]; then
+	if [[ "${1:-}" =~ ^[0-9]{2}:[0-9]{2}$ ]]; then
```

Comment:
Can you explain this change? I would read it as "if ${1} is unset, default to an empty string" but that would be the default behavior without the `:-` part, right?